stub-idp/src/main/java/uk/gov/ida/stub/idp/resources/idp/RegistrationPageResource.java

```diff
@@ -51,34 +58,40 @@
 
 @Path(Urls.REGISTER_RESOURCE)
 @Produces(MediaType.TEXT_HTML)
-@SessionCookieValueMustExistAsASession
+//@SessionCookieValueMustExistAsASession
 public class RegistrationPageResource {
 
     private final IdpStubsRepository idpStubsRepository;
     private final IdpUserService idpUserService;
     private final SamlResponseRedirectViewFactory samlResponseRedirectViewFactory;
     private final NonSuccessAuthnResponseService nonSuccessAuthnResponseService;
     private final SessionRepository<IdpSession> sessionRepository;
+    private final CookieFactory cookieFactory;
+    private final IdpSessionRepository idpSessionRepository;
 
     @Inject
     public RegistrationPageResource(
             IdpStubsRepository idpStubsRepository,
             IdpUserService idpUserService,
             SamlResponseRedirectViewFactory samlResponseRedirectViewFactory,
             NonSuccessAuthnResponseService nonSuccessAuthnResponseService,
-            SessionRepository<IdpSession> sessionRepository) {
+            SessionRepository<IdpSession> sessionRepository,
+            CookieFactory cookieFactory,
+            IdpSessionRepository idpSessionRepository) {
         this.idpUserService = idpUserService;
         this.idpStubsRepository = idpStubsRepository;
         this.samlResponseRedirectViewFactory = samlResponseRedirectViewFactory;
         this.nonSuccessAuthnResponseService = nonSuccessAuthnResponseService;
         this.sessionRepository = sessionRepository;
+        this.cookieFactory = cookieFactory;
+        this.idpSessionRepository = idpSessionRepository;
     }
 
     @GET
     public Response get(
```

Comment:
Maybe add the removed `@SessionCookieValueMustExistAsASession` annotation to this method to enforce sessions for this endpoint but not the pre-register one!

Reply:
I hadn't realised I could annotate the methods with that.  Done.